gobblin-restli/gobblin-job-config-service/server/src/main/java/gobblin/service/JobConfigServer.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobblin.service;
+
+import com.google.common.base.Optional;
+import com.google.common.util.concurrent.Service;
+import com.google.common.util.concurrent.ServiceManager;
+import com.linkedin.r2.filter.FilterChain;
+import com.linkedin.r2.filter.FilterChains;
+import com.linkedin.r2.filter.compression.EncodingType;
+import com.linkedin.r2.filter.compression.ServerCompressionFilter;
+import com.linkedin.r2.transport.common.bridge.server.TransportDispatcher;
+import com.linkedin.r2.transport.http.server.HttpNettyServerFactory;
+import com.linkedin.r2.transport.http.server.HttpServer;
+import com.linkedin.restli.docgen.DefaultDocumentationRequestHandler;
+import com.linkedin.restli.server.DelegatingTransportDispatcher;
+import com.linkedin.restli.server.RestLiConfig;
+import com.linkedin.restli.server.RestLiServer;
+import com.linkedin.restli.server.resources.ResourceFactory;
+import com.typesafe.config.Config;
+import gobblin.runtime.api.MutableJobCatalog;
+import gobblin.runtime.job_catalog.FSJobCatalog;
+import gobblin.util.ConfigUtils;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Server that handles Rest.li requests over HTTP to configure Gobblin Service jobs
+ */
+public class JobConfigServer {
```

Comment:
There is a class being reviewed on another PR that sets up a generic `EmbeddedRestliServer`. I'd prefer if you wait for that to be ready and either make this an extension of that one, or not have this class at all.